chat.sympl

```diff
@@ -518,32 +518,52 @@ def send_message_checks(room_channel: ChannelName, message: str) -> None:
 
 @clientside_helper
 def _get_messages(room_channel: ChannelName) -> List[Message]:
-    # check that the room exists and is accessible
-    room = _get_room(room_channel)
-    room_channel_str : str = room_channel
-    if room.is_deleted:
-        room_channel_str : str = room_channel
-        cvm.error(f'Room {room_channel_str} has been deleted. Cannot get messages.')
-    messages = cvm.storage.query(MessageStatic).in_channel(room_channel).values()
-    def compare(lhs: Message, rhs: Message) -> bool:
-        lt : int = lhs.timestamp
-        return lt < rhs.timestamp
-    return std.sort_by(messages, compare)
+    #gets all the available messages from the room if you have ever been a member, and if the room isnt deleted
+    # if you've never been a member, it throws an error
+    historical_rooms = cvm.storage.query_history(RoomStatic).in_channel(room_channel).values()
+    room = historical_rooms[len(historical_rooms) -1]
+    if isinstance(room, Room):
+        if room.is_deleted:
+            cvm.error(f"Room {room_channel} has been deleted. Cannot get messages.")
+        return cvm.storage.query_history(MessageStatic).in_channel(room_channel).order_by('timestamp', desc=True).values()
```

Comment:
Does this mean someone would have access to messages that were sent before and after the time that they were a member? I know it's just an example, but it seems like limiting the messages to the life of a user's membership would make sense.  But is there a log of members, when they become a member and when the stop being a member? No issue here, just questions out of curiosity.